include/mmtf/map_decoder.hpp

```diff
@@ -263,6 +243,17 @@ void MapDecoder::checkType_(const std::string& key,
                      "entry " << key << std::endl;
     }
 }
+
+template <typename K, typename T>
+void MapDecoder::checkType_(const std::string& key,
+                            msgpack::type::object_type type,
+                            const std::map<K, T>& target) {
+    if (type != msgpack::type::MAP && type != msgpack::type::BIN) {
```

Comment:
```suggestion
    if (type != msgpack::type::MAP) {
```
Binary types would be handled by the BinaryDecoder to deal with any known MMTF encoding. None of those encodings can generate a map and hence binary types shouldn't be ok there.